### PR TITLE
java: add jvm_options to java destination

### PR DESCRIPTION
**Pull request by lbudai**
syslog-ng defines the following options by default:
 * -Djava.class.path
 * -Djava.library.path
 * -Xrs

These java.class.path and java.library.path options are cannot be
overriden from syslog-ng config.

example:
options {
        jvm_options("-Xss1M -XX:+TraceClassLoading");
};

 hdfs(client-lib-dir(/home/stentor/work/hdfs-libs/libs)
      hdfs-uri("hdfs://hdp:8020") hdfs_archive_dir("")
      hdfs_file("data.txt")
      template("$MSG\n")
);

Fixes: #1562

Signed-off-by: Laszlo Budai <laszlo.budai@balabit.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: modules/java/proxies/java-destination-proxy.c [hunk 1]
```diff
@@ -192,14 +192,15 @@ java_destination_proxy_free(JavaDestinationProxy *self)
...
+java_destination_proxy_new(const gchar *class_name, const gchar *class_path, gpointer handle, LogTemplate *template,
+                           const gchar *jvm_options)
 {
   JavaDestinationProxy *self = g_new0(JavaDestinationProxy, 1);
   self->java_machine = java_machine_ref();
   self->formatted_message = g_string_sized_new(1024);
   self->template = log_template_ref(template);
 
-  if (!java_machine_start(self->java_machine))
+  if (!java_machine_start(self->java_machine, jvm_options))
```
Comment (MrAnno) [inline modules/java/proxies/java-destination-proxy.c]: `self->java_machine` points to a "singleton" object, so only the last (or first?) `jvm-options()` usage in the configuration will be in effect.

Shouldn't we use a global option instead?
We can keep this per-destination option as well, but I would not document this in the admin guide due to its misleading behavior.

Comment (lbudai) [inline modules/java/proxies/java-destination-proxy.c]: Yes, it should be a global option.
I'm working on it.

Comment (lbudai) [inline modules/java/proxies/java-destination-proxy.c]: I don't really like it as a global option... but it is better than the per-destination based implementation.

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: modules/java/proxies/java-destination-proxy.c (hunk 1 above)
Comment (MrAnno) [inline modules/java/proxies/java-destination-proxy.c]: Yeah :S It looks weird but this way it's not that deceptive.

**Comment by pzoleex** [comment]
@kira-syslogng test this please test branch=pzolee-java-dest-jvm-options;

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: pzolee-java-dest-jvm-options and test suite: functions

File: modules/java/native/java_machine.c [hunk 2]
```diff
@@ -92,24 +92,92 @@ java_machine_unref(JavaVMSingleton *self)
...
+  static const gchar *predefined_options[] =
+  {
+    "Djava.class.path",
+    "Djava.library.path",
+    NULL
+  };
+
+  for (gint i = 0; predefined_options[i] != NULL; i++)
+    {
+      if (!strncmp(option, predefined_options[i], strlen(predefined_options[i])))
```
Comment (MrAnno) [inline modules/java/native/java_machine.c]: I wouldn't use `strncmp()` when the length of the string can not be known explicitly (`strlen()` won't stop either).

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: pzolee-java-dest-jvm-options and test suite: functions

**Pull request merged**